### ISSUE TITLE
Unix.Unix_error(8, "mkdir", "C:/.../bin/") java build error

### ISSUE THREAD

**Issue by malublu** [HaxeFoundation/haxe#6212]
Haxe Compiler 3.4.2

If i try to build java:
`haxe  -lib wahf -cp src -java C:/.../bin/ -main Main`
i get the error:
`Unix.Unix_error(8, "mkdir", "C:/.../bin/")`

How to fix it? :)

Ps.: since haxe 3.4

**Comment by ibilon** [comment]
Just to be sure, you are on windows? (since it throws a unix_error, but the path is C:)

**Comment by nadako** [comment]
I think OCaml std lib alwayss `Unix_error` for stuff like `mkdir` since it's in the `Unix` module, even though it's compiled for Windows :)

**Comment by nadako** [comment]
@malublu could you add `OCAMLRUNPARAM=b` to the env vars so we see a full stack trace? I _think_ error `8` is `EEXIST` which means that the dir already exists, but it's supposed to check if the path exists before calling `mkdir`...

**Comment by malublu** [comment]
Of course.

https://github.com/HaxeFoundation/haxe/blob/master/src/generators/genjava.ml#L2420

now in the line: https://github.com/HaxeFoundation/haxe/blob/development/src/generators/genjava.ml#L2387

```
Fatal error: exception Unix.Unix_error(8, "mkdir", "C:/.../bin/")
Raised by primitive operation at file "src/generators/genjava.ml", line 2420, characters 51-71
Called from file "src/generators/genjava.ml", line 2421, characters 1-20
Called from file "src/generators/genjava.ml", line 2523, characters 2-15
Called from file "src/main.ml", line 375, characters 2-14
Called from file "src/main.ml", line 848, characters 25-71
Called from file "src/main.ml", line 416, characters 3-11
Called from file "src/main.ml", line 954, characters 1-35
```

works:
`haxe -cp src -java C:/.../bin -main Main`

doesn't works:
```
haxe -cp src -java C:/.../bin/ -main Main
Unix.Unix_error(8, "mkdir", "C:/.../bin/")
```

Sorry i can't speak ocalm to fix it myself :/

But i think it could be simple fixed to remove the trailing slash from the dir before the `Unix.mkdir dir` call like that:

`if(dir.endsWith('/')) dir = dir.substr(0, dir.length - 1);`

**Comment by tobi-one** [comment]
same problem with -cs.

HaxeDevlop defines the "bin\\" dir as output for new java and cs projects .

"%ProgramFiles%\HaxeDevelop\Projects\360 Haxe - C++ Project\Project.hxproj"
"%ProgramFiles%\HaxeDevelop\Projects\365 Haxe - C# Project\Project.hxproj"

**Comment by Jens-G** [comment]
Still not working. Haxe 4.1.5 with PHP target

**Comment by RealyUniqueName** [comment]
Works fine for me. Could you provide a sample?

**Comment by RealyUniqueName** [comment]
Apparently, that's because I'm on linux :)

**Comment by RealyUniqueName** [comment]
The fix will be released with 4.2.1

**Comment by qrp73** [comment]
Catch the same issue in haxe-4.2.3-win.exe

Program.hx:
class Program {
	static function main() {
		var result = 10+20;
		trace('result: $result');
	}
}

haxelib install hxcs
haxe -main Program.hx -cs Program.js

Note: here is mistake I forgot to change extension to *.cs and got this error:
```
Unix.Unix_error(20, "mkdir", "Program.js/src")
```

When I fixed extension to *.cs:
haxe -main Program.hx -cs Program.js

it works fine. But it is not obvious to find mistake from such non-relevant error
